packages/radio-group/src/RadioGroup.tsx

```diff
@@ -84,90 +59,89 @@ interface RadioGroupState {
  * @param props.className classname applied to RadioGroup container.
  * @param props.name Name passed to each Radio belonging to the RadioGroup.
  * @param props.variant Variant to determine if component will appear `default` or `light`.
+ * @param props.size Determines the size of the Radio components Can be 'small' or 'default.
  */
-export default class RadioGroup extends PureComponent<
-  RadioGroupProps,
-  RadioGroupState
-> {
-  static displayName = 'RadioGroup';
-
-  static propTypes = {
-    variant: PropTypes.oneOf(['default', 'light']),
-    className: PropTypes.string,
-    value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-    onChange: PropTypes.func,
-    children: PropTypes.node,
-    name: PropTypes.string,
-  };
-
-  static defaultProps = {
-    variant: Variant.Default,
-    className: '',
-    onChange: () => {},
-  };
-
-  state: RadioGroupState = {
-    value: null,
-  };
+function RadioGroup({
+  variant = Variant.Default,
+  size = Size.Default,
+  className,
+  onChange,
+  children,
+  value: controlledValue,
+  name: nameProp,
+}: RadioGroupProps) {
+  const isControlled = controlledValue != null;
+
+  let defaultChecked = '';
+
+  !isControlled &&
+    React.Children.forEach(children, child => {
+      if (isComponentType(child, 'Radio') && child.props?.default) {
+        defaultChecked = child.props.value;
+      }
+    });
 
-  defaultName = `radio-group-${Math.floor(Math.random() * 1000000)}`;
+  const [uncontrolledValue, setUncontrolledValue] = React.useState<string>(
+    defaultChecked,
+  );
 
-  handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
-    const { onChange, value } = this.props;
+  const name = nameProp || `radio-group-${Math.floor(Math.random() * 1000000)}`;
 
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
     if (onChange) {
       onChange(e);
     }
 
-    if (!value) {
-      this.setState({ value: e.target.value });
+    if (!isControlled) {
+      setUncontrolledValue(e.target.value);
     }
   };
 
-  render() {
-    const {
-      children,
-      name = this.defaultName,
-      className,
-      variant,
-      value = this.state.value,
-    } = this.props;
+  const renderedChildren = React.Children.map(children, (child, index) => {
```

Comment:
We seem to be iterating through children twice in this component. It would be great to consolidate this to one iteration.